resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -217,14 +217,42 @@ private[spark] class ExecutorPodsLifecycleManager(
     ExecutorExited(exitCode, exitCausedByApp, exitMessage)
   }
 
+  // A utility function to try and help people figure out whats gone wrong faster.
+  private def describeExitCode(code: Int): String = {
+    val humanStr = code match {
+      case 0 => "(success)"
+      case 1 => "(generic, look at logs to clarify)"
+      case 42 => "(douglas adams)"
+      // Spark specific
+      case 10 => "(Uncaught exception)"
+      case 50 => "(Uncaught exception)"
+      case 52 => "(JVM OOM)"
+      case 53 => "(DiskStore failed to create temp dir)"
+      // K8s & JVM specific exit codes
+      case 126 => "(not executable - possibly perm or arch)"
+      case 137 => "(SIGKILL, possible container OOM)"
+      case 139 => "(SIGSEGV: that's unexpected)"
+      case 255 => "(exit-1, your guess is as good as mine)"
```

Comment:
I expected we have an error code for `Evicted` due to out of disk during worker decommission. What is the error code for that, @holdenk ?

Reply:
So I think it's going to be inconsistent depending on how exactly it shows up (e.g. does the JVM have an uncaught exception trying to write a file or do we exceed the resource quota). So for now I don't have a clear exit code to map to it unfortunately. I could try and add a base handler for IO errors that are uncaught to exit with a specific code, but I'd rather do that in a separate PR.